recipes/default.rb

```diff
@@ -25,7 +25,7 @@
 nslcd_enable = false
 case node['platform']
 when 'redhat', 'centos', 'scientific'
-  if node.dig('cloud', 'provider')&.downcase == 'ec2'
+  if node.key?('cloud') && node['cloud'].key?('provider') && node['cloud']['provider'].downcase == 'ec2'
```

Comment:
You can use the snippet on `dig()` I gave in my comment if you like, or not.

However, one thing I would suggest changing is avoiding `key?()`. This works equally well and is more succinct:

```
node['cloud'] && node['cloud']['provider'] && node['cloud']['provider'].downcase == 'ec2'
```

That seems to be the more idiomatic way for checking if an attr exists in Chef.